server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -196,11 +202,14 @@ public ClusterState(
         this.metadata = metadata;
         this.routingTable = routingTable;
         this.nodes = nodes;
+        this.transportVersions = Map.copyOf(transportVersions);
```

Comment:
Making this immutable happens in the builder, we don't need this copy right? or only make it immutable here and pass through from the builder?